### PR TITLE
Add Aspire metadata to MongoClient

### PR DESCRIPTION
**Pull request by adelinowona**
## Description

MongoDB drivers are used in various flavors and languages. Making sure we exercise our due diligence in identifying the "origin" of the library calls makes it best to understand how our Atlas servers get accessed.

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [x] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Is this introducing a breaking change?
      - [ ] Yes
        - Link to aspire-docs issue (please use this [`breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)):
      - [ ] No
        - Link to aspire-docs issue (please use this [`doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)):
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6938)

### CODE REVIEW

File: src/Components/Aspire.MongoDB.Driver/AspireMongoDBDriverExtensions.cs [hunk 1]
```diff
@@ -194,6 +194,9 @@ private static MongoClient CreateMongoClient(
         configureClientSettings?.Invoke(clientSettings);
 
         clientSettings.LoggingSettings ??= new LoggingSettings(serviceProvider.GetService<ILoggerFactory>());
+        
+        var aspireVersion = typeof(MongoDBSettings).Assembly.GetName().Version?.ToString();
+        clientSettings.LibraryInfo = new LibraryInfo("aspire", aspireVersion);
```
Comment (eerhardt) [inline src/Components/Aspire.MongoDB.Driver/AspireMongoDBDriverExtensions.cs:199]: Can you move this up to line 188, right after the clientSettings gets created?

The reasoning is that the app developer should be able to set this LibraryInfo in their `configureClientSettings` callback. Doing it here will overwrite anything the app developer tries configuring.

**Review by eerhardt — COMMENTED** [review]
Thanks for the contribution, @adelinowona!

I just had one simple comment. Once that is addressed, I think this can be merged.

File: src/Components/Aspire.MongoDB.Driver/AspireMongoDBDriverExtensions.cs (hunk 1 above)
Comment (adelinowona) [reply src/Components/Aspire.MongoDB.Driver/AspireMongoDBDriverExtensions.cs]: Thanks @eerhardt for raising this, we still want to record the aspire info so I have updated the code to append it if the user has already set the LibraryInfo.

**Comment by adelinowona** [comment]
@eerhardt, @sebastienros, @radical anyone has a chance to review this PR?

**Review by eerhardt — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**